### PR TITLE
Tforkner/2947 investigate backend failing

### PR DESCRIPTION
**Pull request by thetif**
issue #2971 

The backend is randomly failing after a random period of time. I thought that upgrading things might improve the stability. I upgraded node to node 14 on the back and front end. I upgraded all of the backend dependencies. And I updated most of the frontend dependencies (except the ones related to React 17 and webpack 5). I will work on those at a later time.

### This pull request changes...

- updates node to 14 on all servers
- updated dependencies on the backend
- updated dependencies on the front end
- updates code as needed to work with new dependencies

### This is how to verify this change...
- verify that tests pass
- verify that the applications still operates as expected

### This pull request is ready to merge when...

- [x] Develop has updated automated tests (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] QA has manually tested and created bugs for any issues found (beyond minor fixes)
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] The change has been documented
- [x] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by tbolt — APPROVED** [review]
I put a note about the node engine and some other minor things, otherwise looks good. 

I think this would be good to have @eStar386 do a thorough test of the app as some of these package changes could break things.

File: api/package.json [hunk 1]
```diff
@@ -68,60 +68,49 @@
...
-    "zxcvbn": "^4.4.2"
+    "pg": "^8.5.1",
+    "sanitize-html": "^2.3.2",
+    "sharp": "^0.27.2",
+    "uuid": "^8.3.2",
+    "winston": "^3.3.3"
   },
   "engines": {
-    "node": "^10.14"
+    "node": "^12.14.0"
```
[the remark this answers is not in the document]
Comment (thetif) [reply api/package.json:95]: yes, thanks, sorry, I did an incremental update to 12 and then to 14 and I must have missed that one. I'll fix it.

File: api/routes/docs/get.test.js [hunk 2]
```diff
@@ -1,39 +1,28 @@
...
+const mockExpress = require('../../util/mockExpress');
+const mockResponse = require('../../util/mockResponse');
 
-  const di = {
-    getFile: sandbox.stub()
-  };
+let app;
+let res;
+let next;
+let getFile;
```
[the remark this answers is not in the document]
Comment (thetif) [reply api/routes/docs/get.test.js:12]: yeah I think the linter complains about that

File: web/src/util/formats.test.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ describe('formatting util', () => {
 
   test('formats percents', () => {
     expect(formatPerc(0.2575)).toEqual('+26%');
-    expect(formatPerc(-0.35)).toEqual('-35%');
+    expect(formatPerc(-0.35)).toEqual('−35%');
```
[the remark this answers is not in the document]
Comment (thetif) [reply web/src/util/formats.test.js:22]: haha, yeah I know, I couldn't see what the difference was, but the tests did :D

File: api/routes/docs/get.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (thetif) [reply api/routes/docs/get.test.js:12]: yeah double checked that and eslint does not like it

**Comment by eStar386** [comment]
@thetif Can you update this with the pull request in action link?

**Comment by eStar386** [comment]
Creating the test automation suite by transforming the manual tests into an automated regression test suite.

**Comment by eStar386** [comment]
Per a slack discussion with @thetif - it was determined that the automation efforts shouldn't be attached to tickets due to  release requirements. 

Regression Testing: Passed

**Review by eStar386 — APPROVED** [review]

**Pull request merged**